### PR TITLE
silo: variant python needs python

### PR DESCRIPTION
**Pull request by cessenat**
Variant "python" requires Python.h which is provided by the "python" package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/silo/package.py [hunk 1]
```diff
@@ -129,6 +130,8 @@ def flag_handler(self, name, flags):
             if spec.satisfies("%oneapi"):
                 flags.append("-Wno-error=int")
                 flags.append("-Wno-error=int-conversion")
+            if "+python" in spec:
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/silo/package.py]:
```suggestion
            if spec.satisfies("+python"):
```
(Fewer instantiations.)

File: var/spack/repos/builtin/packages/silo/package.py [hunk 2]
```diff
@@ -129,6 +130,8 @@ def flag_handler(self, name, flags):
...
+                flags.append("-I {0}".format(spec["python"].headers.directories[0]))
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/silo/package.py]:
```suggestion
                flags.append(f"-I {spec['python'].headers.directories[0]}")
```
F-strings preferred in new code.

File: var/spack/repos/builtin/packages/silo/package.py [hunk 3]
```diff
@@ -65,6 +65,7 @@ class Silo(AutotoolsPackage):
     variant("hzip", default=True, description="Enable hzip support")
     variant("fpzip", default=True, description="Enable fpzip support")
 
+    depends_on("python", type="build", when="+python")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/silo/package.py]: Does this not need python at run time? Why does it need python.h if not to then link against it?

See https://github.com/LLNL/Silo/blob/f57b12f1faaf9699ad86a60602cd9a65689585cc/tools/python/CMakeLists.txt#L69

```suggestion
    depends_on("python", type=("build", "link"), when="+python")
```
And move to where other "regular" dependencies are.

**Comment by wdconinc** [comment]
Waiting for approval from maintainer before merging.

**Review by patrickb314 — APPROVED** [review]
Looks good to me. Merge approved.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**